substrate/client/network/transactions/src/config.rs

```diff
@@ -57,13 +57,13 @@ pub type TransactionImportFuture = Pin<Box<dyn Future<Output = TransactionImport
 /// Transaction pool interface
 pub trait TransactionPool<H: ExHashT, B: BlockT>: Send + Sync {
 	/// Get transactions from the pool that are ready to be propagated.
-	fn transactions(&self) -> Vec<(H, B::Extrinsic)>;
+	fn transactions(&self) -> Vec<(H, Arc<B::Extrinsic>)>;
 	/// Get hash of transaction.
-	fn hash_of(&self, transaction: &B::Extrinsic) -> H;
+	fn hash_of(&self, transaction: &Arc<B::Extrinsic>) -> H;
```

Comment:
I think Arc is not needed here.

Reply:
I mean we are already passing reference to the extrinsics data.

Reply:
Yeah :+1: